app/services/fetch_github_pull_request_data_service.rb

```diff
@@ -0,0 +1,97 @@
+class FetchGithubPullRequestDataService
+  attr_reader :pull_request_url, :client
+  GITHUB_BASE_URL = "https://api.github.com"
+
+  def initialize(pull_request_url)
+    @pull_request_url = pull_request_url
+    @client = Github::Client.new
+  end
+
+  def call
+    data = parse_pull_request_url
+    pr_data = fetch_pull_request_data(data)
+
+    return unless pr_data
```

Comment:
No error handling?

Reply:
Yeah, i think it will be good to add some better error handling, i will do